lib/src/deploy.rs

```diff
@@ -560,7 +587,11 @@ pub(crate) async fn stage(
     image: &ImageState,
     spec: &RequiredHostSpec<'_>,
     prog: ProgressWriter,
+    StageOptions { deploy_kexec }: StageOptions,
 ) -> Result<()> {
+    let steps_total = 4 + (deploy_kexec as u64);
+    let mut steps = 0;
```

Comment:
How about:
```
let mut steps = 4;
if deploy_kexec {
  steps += 1;
}
```
?

Feels like it will "scale" a bit more cleanly if we have more conditionals later.

(This all said, I'd really like to have higher level wrappers for the progress API because the numbers here are SO easy to mess up)

Reply:
Yeah, that's pretty much the same idea, but I see how that would be cleaner.